com.vrcfury.vrcfury/Editor/VF/Feature/BakeHapticSocketsBuilder.cs

```diff
@@ -233,6 +248,29 @@ public class BakeHapticSocketsBuilder : FeatureBuilder {
                             socket.unitsInMeters
                         );
                     }
+
+                    if (socket.IsValidPlugLength) {
+                        // Calculate `Plug Length` using `TPS_Pen_Penetrating - TPS_Pen_Root + 0.01` 
+                        var lastPlugLength = fx.NewFloat("LastPlugLength");
+                        var plugLength = math.Subtract(penTip, penRoot);
+                        directTree.Add(math.MakeSetter(plugLength, 0.01f));
+                        var validWhen = math.And(math.GreaterThan(penRoot, 0.001f), math.LessThan(penTip, 0.999f));
+                        var plugLengthValid = math.SetValueWithConditions("ValidPlugLength", (plugLength, validWhen), (lastPlugLength, null));
+                        var plugLengthGlobal = fx.NewFloat(socket.plugLengthParameterName, usePrefix: false);
+                        directTree.Add(math.MakeCopier(plugLengthValid, plugLengthGlobal));

```

Comment:
MakeCopier introduces one frame of lag. You can save this by just using the global directly.